src/mds/Server.cc

```diff
@@ -3779,7 +3779,6 @@ void Server::handle_client_setattr(MDRequestRef& mdr)
       mds->locker->drop_locks(mdr.get());
       mdr->drop_local_auth_pins();
       cur->add_waiter(CInode::WAIT_TRUNC, new C_MDS_RetryRequest(mdcache, mdr));
-      mds->mdlog->flush();
```

Comment:
I think we need to leave this flush in. Adding in an immediate flush once it's logged doesn't matter so much, but in this case isn't the client waiting for the update to be durable before it does anything?